File: p3.cpp
/*
 * p3.cpp
 *
 *  Created on: Mar 21, 2015
 *      Author: alex
 */
#include <iostream>
#include <math.h>

using namespace std;

int main(){
	long long num = 999999999999999999;
	for (long long i = 2; i <= sqrt(num); i++){
		while(num % i == 0){
			num /= i;
			cout << i << endl;
		}
	}
	cout << num;
}

File: p3.java
package problems;

public class p3 {
   static long num = 1008514751436008514L;

   public static void main(String[] args) {
      Long b = System.nanoTime();
      long a = find(num);
      System.out.println((System.nanoTime() - b) / 1000000);
      System.out.println(a);
   }

   private static long find(long val) {
      long root = (long) Math.sqrt(val);
      for (int i = 2; i <= root; i++) {
         if (val % i == 0) {
            return Math.max(i, find(val / i));
         }
      }
      return val;
   }
}